devito/finite_differences/derivative.py

```diff
@@ -307,7 +307,8 @@ def _xreplace(self, subs):
         if self in subs:
             new = subs.pop(self)
             try:
-                return new._xreplace(subs)
+                new, flag = new._xreplace(subs)
+                return new, True
```

Comment:
Note: this was actually incorrectly returning `False` flag in lots of cases (since subs is then empy here) so `new._xreplace(subs)` would return `new, False`